force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,151 @@
+public with sharing class OpportunityHelper {
+    public List<Opportunity> newRecords;
+    public List<Opportunity> oldRecords;
+    public Map<Id,Opportunity> newRecordsMap;
+    public Map<Id,Opportunity> oldRecordsMap;
+
+    public OpportunityHelper() {
+        this.newRecords = Trigger.new;
+        this.oldRecords = Trigger.old;
+        this.newRecordsMap = (Map<Id, Opportunity>) Trigger.newMap;
+        this.oldRecordsMap = (Map<Id, Opportunity>) Trigger.oldMap;
+    }
+
+    ///OpportunityTrigger.trigger implementation  ===== Opportunity Part 1
+
+    public static void oppAmountValidation (List<Opportunity> newRecords) {
+        for(Opportunity opp : newRecords){
+            if(opp.Amount < 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+    }
+    public static void closedWonOppDeletion(List<Opportunity> oldRecords){
+        Map<Id, Account> accounts = new Map<Id, Account>([SELECT Id, Industry FROM Account WHERE Id IN (SELECT AccountId FROM Opportunity WHERE Id IN :oldRecords)]);
+        for(Opportunity opp : oldRecords){
+            if(opp.StageName == 'Closed Won'){
+                if(accounts.get(opp.AccountId).Industry == 'Banking'){
+                    opp.addError('Cannot delete a closed won opportunity for a banking account');
+                }
+            }
+        }
+    }
+
+    public static void assignPrimaryContactWithTitleCEO(List<Opportunity> newRecords){
+        Set<Id> accountIds = new Set<Id>();
+        for(Opportunity opp : newRecords){
+            accountIds.add(opp.AccountId);
+        }
+        Map<Id, Contact> contacts = new Map<Id, Contact>([SELECT Id, FirstName, AccountId, Title FROM Contact WHERE AccountId IN:accountIds AND Title = 'CEO' ORDER BY FirstName Asc]);
+        Map<Id, Contact> accountIdToContacts = new Map<Id,Contact>();
+        for(Contact con : contacts.values()){
+            if(!accountIdToContacts.containsKey(con.AccountId)){
+                accountIdToContacts.put(con.AccountId, con);
+            }
+        }
+        for(Opportunity opp : newRecords){
+            if(opp.Primary_Contact__c == null){
+                if(accountIdToContacts.containsKey(opp.AccountId)){
+                    opp.Primary_Contact__c = accountIdToContacts.get(opp.AccountId).Id;
+                }
+                
+            }
+        }
+
+    }
```

Comment:
good. One thing is the Maps are unordered. So no real need to order your query by name.